packages/typegpu/src/core/slot/accessor.ts

```diff
@@ -0,0 +1,81 @@
+import type { AnyWgslData } from '../../data';
+import type { Exotic } from '../../data/exotic';
+import { getResolutionCtx } from '../../gpuMode';
+import type { Infer } from '../../shared/repr';
+import type { ResolutionCtx } from '../../types';
+import { type TgpuBufferUsage, isBufferUsage } from '../buffer/bufferUsage';
+import { type TgpuFn, isTgpuFn } from '../function/tgpuFn';
+import { slot } from './slot';
+import type { TgpuAccessor, TgpuSlot } from './slotTypes';
+
+// ----------
+// Public API
+// ----------
+
+export function accessor<T extends AnyWgslData>(
+  schema: T,
+  defaultValue?:
+    | TgpuFn<[], Exotic<T>>
+    | TgpuBufferUsage<Exotic<T>>
+    | Infer<Exotic<T>>,
+): TgpuAccessor<Exotic<T>> {
+  return new TgpuAccessorImpl(schema as Exotic<T>, defaultValue);
+}
+
+// --------------
+// Implementation
+// --------------
+
+export class TgpuAccessorImpl<T extends AnyWgslData>
+  implements TgpuAccessor<T>
+{
+  readonly resourceType = 'accessor';
+  public label?: string | undefined;
+  public slot: TgpuSlot<TgpuFn<[], T> | TgpuBufferUsage<T> | Infer<T>>;
+
+  constructor(
+    public schema: T,
+    public defaultValue:
```

Comment:
I think those could be readonly (I know they are on the interface but just to be consistent)